tests/test_json_schema.py

```diff
@@ -5149,3 +5164,23 @@ class ModelTest(BaseModel):  # noqa: F811
     # call the __get_pydantic_json_schema__ method multiple times)
     # but it's much easier to test for than absence of a recursion limit
     assert calls == 1
+
+
+def test_callable_json_schema_extra():
+    def pop_default(s):
+        s.pop('default')
+
+    class Model(BaseModel):
+        a: int = Field(default=1, json_schema_extra=pop_default)
+        b: Annotated[int, Field(json_schema_extra=pop_default)] = 2
+        c: Annotated[int, Field(json_schema_extra=pop_default), Field(default=3)]
+        d: Annotated[int, Field(default=4), Field(json_schema_extra=pop_default)]
+        # TODO: it doesn't work properly to have both annotation and assigned value of FieldInfo:
+        # e: Annotated[int, Field(json_schema_extra=pop_default)] = Field(default=5)
+        # f: Annotated[int, Field(default=6)] = Field(json_schema_extra=pop_default)
```

Comment:
I think it would be possible to make this and the previous case work (i.e., `e` and `f`), but I tried and it ended up being a significantly bigger change and I wasn't able to see it _all_ the way through and get all tests passing. So rather than wasting time on it for now, I just added a note here. Happy to create an issue for this if there isn't one already. (CC @adriangb)

While it might be tempting to try to make this an error, for various reasons, I think it would be preferable if assigning a `FieldInfo` was equivalent to adding it as the last annotation on the field, which _does_ work as shown above.

Reply:
I do think we should make it work at some point. Is it not working now? I thought it was.

Reply:
It works for most things, it just doesn't work for `json_schema_extra` and `default` specifically, annoyingly.

Reply:
Also, even on main, this doesn't work:
```python
from pydantic import BaseModel, Field

from typing_extensions import Annotated

class Model(BaseModel):
    a: Annotated[int, Field(default=1)] = Field()
Model()
```
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for Model
a
  Field required [type=missing, input_value={}, input_type=dict]
```

Reply:
I was able to fix this through an appropriate change in `FieldInfo.from_annotated_attribute` 💪